### ISSUE TITLE
Field.getDataType(Configuration) resets nullability

### ISSUE THREAD

**Issue by rbuehlma** [jOOQ/jOOQ#11334]
### Expected behavior

nullable() should be false in the below sample

### Actual behavior

nullable() is true

### Steps to reproduce the problem

```
    @Test
    public void testCreateTable() throws Exception {
        Connection conn = getSourceDbConnection();
        conn.createStatement().execute("create table test_table (col5 varchar not null)");
        DSLContext dsl = DSL.using(conn);
        org.jooq.Table<?> testTable = dsl.meta().getSchemas("test-source").get(0).getTable("test_table");
        Field field = testTable.field(0);
        DataType dataType = field.getDataType(dsl.configuration());
        assertThat(dataType.nullable(), Matchers.is(false));
    }
```

### Versions

 - jOOQ: 3.14.6
 - Java: 11
 - Database (include vendor): Postgresql 10.7
 - OS: Linux
 - JDBC Driver (include name if inofficial driver): org.postgresql (42.2.5)

**Comment by lukaseder** [comment]
Thanks a lot for your report. Wow, how did that slip by integration tests! I'll look into it immediately.

Notice, you can do

```java
dsl.execute("create table test_table (col5 varchar not null)");
```

**Comment by lukaseder** [comment]
Oh, I see. The bug is not related to your usage of `Meta`. It can be reproduced like this as well:

```java
Field<Integer> f = DSL.field("x", INTEGER.nullable(false));
System.out.println(f.getDataType().nullable());
System.out.println(f.getDataType(ctx.configuration()).nullable());
```

Printing:

```
false
true
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.7 (#11337)
